From: Gregor Jasny <gjasny@googlemail.com>
Subject: Disable fancy upsampling (needed for libjpeg >= 7)
Bug-Debian: http://bugs.debian.org/647273
Forwarded: yes

Libv4lconvert expects to receive downsampled data. With
jpeglib 7 the default changed to TRUE. So we have to
explicitely request downsampled data after reading the
JPEG header.
Thanks to Guido Vollbeding for pointing this out.

Signed-off-by: Gregor Jasny <gjasny@googlemail.com>

--- a/lib/libv4lconvert/jpeg.c
+++ b/lib/libv4lconvert/jpeg.c
@@ -390,6 +390,7 @@
 		}
 
 		data->cinfo.raw_data_out = TRUE;
+                data->cinfo.do_fancy_upsampling = FALSE;
 		jpeg_start_decompress(&data->cinfo);
 		/* Make libjpeg errors report that we've got some data */
 		data->jerr_errno = EPIPE;
